specification/schemaregistry/SchemaRegistry/client.tsp

```diff
@@ -0,0 +1,62 @@
+import "@azure-tools/typespec-client-generator-core";
+import "@typespec/versioning";
+import "./main.tsp";
+
+using TypeSpec.Versioning;
+using Azure.ClientGenerator.Core;
+using SchemaRegistry;
+
+@TypeSpec.Versioning.useDependency(
+  ServiceApiVersions.V2023_07_01
+)
+namespace ClientCustomizations;
+
+@client({name: "SchemaRegistryClient", service: SchemaRegistry})
+interface SchemaOperations {
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard read operation w/ SchemaId resource doesn't allow returning binary schema and headers."
+  listSchemaGroups is SchemaRegistry.listSchemaGroups;
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard read operation w/ Version resource doesn't allow returning non-version response."
+  listSchemaVersions is SchemaRegistry.listSchemaVersions;
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard read operation w/ SchemaId resource doesn't allow returning binary schema and headers."
+  getSchemaById is SchemaRegistry.getSchemaById;
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard read operation w/ Version resource doesn't allow returning non-version response."
+  getSchemaByVersion is SchemaRegistry.getSchemaByVersion;
+  #suppress "@azure-tools/typespec-azure-core/byos" "storage service is not used"
+  getSchemaPropertiesByContent is SchemaRegistry.getSchemaPropertiesByContent;
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard createOrReplace operation does not allow passing in binary body to request."
+  #suppress "@azure-tools/typespec-azure-core/byos" "storage service is not used"
+  registerSchema is SchemaRegistry.registerSchema;
+}
+
+@@access(SchemaGroup, Access.internal);
+
+// csharp modifications
+@doc("The content type for the schema.")
+enum ContentTypeEnum {
+  @doc("The Avro schema content type.")
+  avro: "application/json; serialization=Avro",
+
+  @doc("The JSON schema content type.")
+  @added(ServiceApiVersions.V2022_10)
+  json: "application/json; serialization=Json",
+
+  @doc("The custom schema content type.")
+  @added(ServiceApiVersions.V2022_10)
+  custom: "text/plain; charset=utf-8",
+
+  @doc("The protobuf schema content type.")
+  @added(ServiceApiVersions.V2023_07_01)
+  protobuf: "text/vnd.ms.protobuf",
+}
+
+@@access(ContentTypeEnum, Access.internal);
+@@usage(ContentTypeEnum, Usage.input | Usage.output);
+@@projectedName(ContentTypeEnum, "csharp", "ContentType");
+
+// operations
+@@access(listSchemaGroups, Access.internal);
```

Comment:
Shouldnt these be accessed through the interface, I'm not sure this will always behave as expected without the interface specified:
```suggestion
@@access(SchemaOperations.listSchemaGroups, Access.internal);
```

Reply:
When I generate with the interface specified, it doesn't seem to pick up the ones in main and so groups the operations. So when we generate, it outputs a public operations folder, which we don't want.

Reply:
not sure if that's a bug, but I can also just repeat this and make the operations internal, both with and without SchemaOperations specified.

Reply:
I think that what is happening here is that the original operation is marked as internal access and that is carried over to the ones defined in the client customizations....I suppose it is fine to leave as is in that case but if that ever changes then you would have to specify the operations on the customization interface as internal.